data_structures/reverse_a_linked_list.cpp

```diff
@@ -163,6 +181,41 @@ int32_t list::traverse(int32_t index) {
     exit(1);
 }
 
+/**
+ *  @brief calls delete operator on every node in the represented list
+ */
+void list::delete_all_nodes() {
+    while (head != nullptr) {
+        const auto tmp_node = head->next;
+        delete head;
+        head = tmp_node;
+    }
+}
+
+list::~list() { delete_all_nodes(); }
+
+void list::copy_all_nodes_from_list(const list& other) {
```

Comment:
There's no documentation here.
Also, I think you missed the documentation of the parameters. 🤔

Reply:
I think that the destructor does not need to be documented. Similarly the arguments of the copy constructor and the assignment operator. It is well-defined what are these functions and what should they do.

I did not document the function `list::copy_all_nodes_from_list` because it is private (i.e. the user will not need its documentation) and really short.

Reply:
If the users are trying to understand and learn your algorithm (which is the purpose of this repository), without documentation, it's going to be harder for them to do so. Please document accordingly.